app/lua/ltable.c

```diff
@@ -580,7 +580,7 @@ const TValue *luaH_getnum (Table *t, int key) {
 
 /* same thing for rotables */
 const TValue *luaH_getnum_ro (void *t, int key) {
-  const TValue *res = luaR_findentryN(t, key, NULL);
+  const TValue *res = NULL;  // integer values not supported: luaR_findentryN(t, key, NULL);
```

Comment:
Do we get a compile-time error if someone does try to register a numeric key in an RO table? If not, can we?

Reply:
There is no macro to do this, so they can't.  None of our current code uses numerically indexed ROTables, and there is roughly a 2× performance hit in leaving in this feature that we don't use, and so I removed it.

If there we want to add it back we should think about how we could do this but we should have this as a discussion on the issue, as there are trade-offs here.